types/Database/Eloquent/ModelNotFoundException.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use function PHPStan\Testing\assertType;
+
+/** @var ModelNotFoundException<User> $exception */
+$exception = new ModelNotFoundException();
```

Comment:
Constructing this class generically is always going to be awkward, since the generic type can not be derived from the constructor arguments. Perhaps it can be considered to move the arguments that are currently passed to `setModel()` into the constructor for the next major version of Laravel?